src/util/are_part_min_max_increasing.py

```diff
@@ -0,0 +1,87 @@
+from collections import defaultdict
+
+import pyarrow.parquet as pq
+import argparse
+
+from urllib.parse import urlparse
+import boto3
+import gzip
+from tqdm.auto import tqdm
+
+
+def are_parquet_file_row_groups_min_max_ordered(pf: pq.ParquetFile, column_name: str) -> bool:
+    sort_column_index = next(i for i, name in enumerate(pf.schema.names)
+                             if name == column_name)
+
+    # keep track of min/max in this ParquetFile
+    whole_min = None
+    whole_max = None
+    prev_max = None
+    for row_group_index in range(pf.num_row_groups):
+        row_group = pf.metadata.row_group(row_group_index)
+        column = row_group.column(sort_column_index)
+        if prev_max is not None and prev_max > column.statistics.min:
```

Comment:
Must prepared that no min/max statistics are available in a row group because of overlong URLs / SURT keys. Cf. [PARQUET-1685](https://issues.apache.org/jira/browse/PARQUET-1685):

```
    if prev_max is not None and prev_max > column.statistics.min:
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: '>' not supported between instances of 'str' and 'NoneType'
```

Seen on <https://data.commoncrawl.org/cc-index/table/cc-main/warc/crawl=CC-MAIN-2022-33/subset=crawldiagnostics/part-00272-d466b69e-be2b-4525-ac34-1b10d57329da.c000.gz.parquet>